examples/SDG1.ipynb

```diff
@@ -0,0 +1,1336 @@
+        "cluster_topic_mapping, new_topics"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "CX26-PGdcui0"
+      },
+      "source": [
+        "And finally we can use this information to further prompt a model to keep generating synthetic data. We do this by passing all the topics in the list of jsons to the prompt below."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "zHf4LnVk0aHw"
+      },
+      "outputs": [],
+      "source": [
+        "output_string = \"\"\n",
+        "for i in range(3):\n",
+        "  question = f\"\"\"\n",
+        "  I am creating input output training pairs to fine tune my gpt model. I want the input to be product name and category and output to be description. the category should be things like: mobile phones, shoes, headphones, laptop, electronic toothbrush, etc. and also more importantly the categories should come under some main topics: {[entry['topic'] for entry in cluster_topic_mapping]})\n",
+        "  After the number of each example also state the topic area. The format should be of the form:\n",
+        "  1. topic_area\n",
+        "  Input: product_name, category\n",
+        "  Output: description\n",
+        "\n",
+        "  Do not add any extra characters around that formatting as it will make the output parsing break.\n",
+        "\n",
+        "  Here are some helpful examples so you get the style of output correct.\n",
+        "\n",
+        "  1) clothing\n",
+        "  Input: \"Nike Air Max 270, Shoes\"\n",
+        "  Output: \"Experience unparalleled comfort with the Nike Air Max 270. These shoes feature a blend of modern style and the traditional Air Max cushioning, perfect for those always on the move.\"\n",
+        "  \"\"\"\n",
+        "\n",
+        "  response = client.chat.completions.create(\n",
+        "    model=\"gpt-4\",\n",
+        "    messages=[\n",
+        "      {\"role\": \"system\", \"content\": \"You are a helpful assistant designed to generate synthetic data.\"},\n",
+        "      {\"role\": \"user\", \"content\": question}\n",
+        "    ]\n",
+        "  )\n",
+        "  res = response.choices[0].message.content\n",
+        "  output_string += res + \"\\n\" + \"\\n\"\n",
+        "print(output_string)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RQMHQxnZdRug"
+      },
+      "source": [
+        "You can run this in a loop to append to your previous data and in this way you can keep generating more textual synthetic data to train another GPT model while making sure that we cater to imbalanced datasets and generating a diversity of data."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Hiim8Xg5djGH"
+      },
+      "source": [
+        "You have now completed part 1 of the synthetic data generation tutorial where we have gone through:\n",
```

Comment:
how do you think developers should validate/test against the quality of the synthetic data and what are the common pitfalls? would be good to add some guidance here.

Reply:
I am adding that in part 2 as part of the prompt engineering techniques. how to check for bad data.